### ISSUE TITLE
Process and alignment completed successfully, but .tre file empty

### ISSUE THREAD

**Issue by smdabdoub** [AstrobioMike/GToTree#29]
Hi Mike,

Thanks for all your effort on GToTree! I seem to be running into an issue, however. I'm aligning (what ends up being) 1208 genomes from NCBI assemblies using the Bacteria HMM. The log file reports nothing unusual, and the Aligned_SCGs.faa appears to be fine, but the tree file is 0 bytes.

I ran fasttree manually on the Aligned_SCGs file and it completed with a non-empty tree file, so I'm not sure what happened.

The fasttree output from the original GToTree run is below:

```
FastTree Version 2.1.10 Double precision (No SSE3)
Alignment: gtotree_homd/Aligned_SCGs.faa
Amino acid distances: BLOSUM45 Joins: balanced Support: SH-like 1000
Search: Normal +NNI +SPR (2 rounds range 10) +ML-NNI opt-each=1
TopHits: 1.00*sqrtN close=default refresh=0.80
ML Model: Jones-Taylor-Thorton, CAT approximation with 20 rate categories
Ignored unknown character X (seen 420498 times)
Initial topology in 28.53 seconds of   1149   152 seqs (at seed   1000)   
Refining topology: 41 rounds ME-NNIs, 2 rounds ME-SPRs, 20 rounds ML-NNIs
Total branch-length 37.141 after 144.56 sec1101 of 1150 splits, 0 changes   ax delta 0.003)    
ML-NNI round 1: LogLk = -2278017.783 NNIs 219 max delta 230.14 Time 455.01 (max delta 230.138)   
Switched to using 20 rate categories (CAT approximation)20 of 20   
Rate categories were divided by 1.115 so that average rate = 1.0
CAT-based log-likelihoods may not be comparable across runs
Use -gamma for approximate but comparable Gamma(20) log-likelihoods
ML-NNI round 2: LogLk = -2091995.647 NNIs 105 max delta 85.56 Time 725.04s (max delta 85.563)   
ML-NNI round 3: LogLk = -2091897.565 NNIs 62 max delta 9.96 Time 790.76s (max delta 9.957)   
ML-NNI round 4: LogLk = -2091886.578 NNIs 34 max delta 8.27 Time 827.64s (max delta 8.272)   
ML-NNI round 5: LogLk = -2091886.574 NNIs 7 max delta 0.00 Time 838.75s (max delta 0.000)   
Turning off heuristics for final round of ML NNIs (converged)
ML-NNI round 6: LogLk = -2090748.848 NNIs 51 max delta 8.23 Time 1069.27 (final)elta 8.230)   
Optimize all lengths: LogLk = -2090741.662 Time 1138.45
   1187.78 seconds: ML split tests for    300 of   1149 internal splits  
```

and the tail of the runlog is:

```
#################################################################################
####                                 [0;32mDone!![0m                                  ####
#################################################################################

  Overall, 1208 genomes of the input 1352 were retained (see notes below).

    Tree written to:
        [0;32mgtotree_homd/gtotree_homd.tre[0m

    Alignment written to:
        [0;32mgtotree_homd/Aligned_SCGs.faa[0m

    Main genomes summary table with comp./redund. estimates written to:
        [0;32mgtotree_homd/Genomes_summary_info.tsv[0m

    Summary table with hits per target gene per genome written to:
        [0;32mgtotree_homd/SCG_hit_counts.tsv[0m

    Partitions (for downstream use with mixed-model treeing) written to:
        [0;32mgtotree_homd/run_files/Partitions.txt[0m

 _______________________________________________________________________________

  Notes:

        4 accession(s) redundant.
        135 accession(s) not successfully found at NCBI.
        5 genome(s) removed due to having too few hits to the targeted SCGs.
        2 genome(s) had an estimated redundancy of >= 10%.

    Reported along with additional informative run files in:
        [0;32mgtotree_homd/run_files/[0m

 _______________________________________________________________________________

    Log file written to:
        [0;32mgtotree_homd/gtotree-runlog.txt[0m

    Programs used and their citations have been written to:
        [0;32mgtotree_homd/citations.txt[0m


                                         Total process runtime: 3 hours and 7 minutes.
                                                      [0;32mHappy Thursday :)[0m


```

Thanks!

**Comment by AstrobioMike** [comment]
Hey there, Shareef!

That’s definitely not good, ha. Thank you for writing in so I can look into it :)

Would you mind attaching the full GToTree run log file here? Or emailing it to me if you’d prefer (MikeLee@bmsis.org)?

**Comment by smdabdoub** [comment]
Absolutely!

[runlog.txt](https://github.com/AstrobioMike/GToTree/files/5928965/runlog.txt)

**Comment by AstrobioMike** [comment]
Thanks, Shareef!

I hate when this happens, but I can't seem to recreate the problem, neither with v1.5.38 (which was used in the log file), or the latest version, on either mac or linux :(

I realize there may be no need for you to run things again, but if you did happen to run it again, could you try with the latest version of GToTree and let me know if you hit the same issue? It could be installed with:

```
conda create -n gtotree-1.5.46 -c conda-forge -c bioconda -c defaults -c astrobiomike gtotree=1.5.46
conda activate gtotree-1.5.46
```
(to make sure it's installing the latest one in case anything weird is going on with that)

You also made me realize something else though. Taxonomy information isn't added if we don't specify the `-t` flag (for NCBI taxonomy) or the `-D` flag for GTDB taxonomy, even if we specify which ranks we want with the `-L` flag, but I wasn't previously alerting the user to this. So even though you included `-L Genus,Species`, no tax info was being added to the labels in the output. This was an oversight on my part, and now the program tells us up-front if we specified an `-L` argument, but we didn't give either the `-t` or `-D` flags. So we don't have to go through the entire run now just to find out it didn't actually add any taxonomic info 👍  

So that's at least 1 thing you helped me improve by writing in about this regardless of me not being able to recreate the empty tree-file issue :) Thanks!

**Comment by smdabdoub** [comment]
I actually ended up running everything again with v1.5.38 and it worked fine, so....cosmic rays? I'll re-run with the latest version as well to check.

Good to know RE adding taxonomy! Happy to help, even indirectly!

**Comment by AstrobioMike** [comment]
haha! Thanks for letting me know it worked out :)

**Comment by kellikmullane** [comment]
Hey there! I know this is a closed issue, but I'm having the exact same problem. My run completes successfully and gives no errors/flags, but anytime I use the -m flag the .tre file is empty at the end of the run. It seems to be recognizing the file associated with the -m flag and is utilizing it, as the labels are included in the 'Genomes_summary_info.tsv' output file. 

As input, I'm providing a .txt file with paths to each genome and using the -f flag. I've tried passing both a .tsv and .txt file with the -m flag. I saw the note under the -m argument explainer on Github saying the file being passed to -m needs to include all of the input genomes - I double checked this and they are all included.  

I recently downloaded GToTree (running v1.6.37), and have been running it on a smaller subset of my dataset (full dataset takes ~15 hours to run on my machine). Would REALLY like to get this working for the full dataset. Would save me so much time and energy. 

I've attached the runlog here for your reference. 

Appreciate any help/advice you have!
[gtotree-runlog.txt](https://github.com/AstrobioMike/GToTree/files/9552676/gtotree-runlog.txt)
